src/dal-plugin/PlugIn.mm

```diff
@@ -175,6 +175,11 @@ - (void)setPropertyDataWithAddress:(CMIOObjectPropertyAddress)address qualifierD
 - (void)receivedFrameWithSize:(NSSize)size timestamp:(uint64_t)timestamp frameData:(nonnull NSData *)frameData {
     dispatch_sync(_stateQueue, ^{
         if (_state == PlugInStateWaitingForServer) {
+            dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Fine to remove the dispatch_async here. You're right; without `synchronize` this really isn't necessary